zh-learn-cli/src/main/java/com/zhlearn/cli/audio/InteractiveAudioUI.java

```
@@ -0,0 +1,150 @@
+        }
+    }
+
+    private void renderAnsi(java.io.PrintWriter writer, SelectionSession session) {
+        // Clear screen and move cursor to home; hide cursor
+        writer.print("\u001B[2J\u001B[H\u001B[?25l");
+        writer.println();
+        for (int i = 0; i < session.size(); i++) {
+            PronunciationCandidate c = session.candidateAt(i);
+            boolean selected = (i == session.currentIndex());
+            String marker = selected ? ">" : " ";
+            String line = String.format("%s %2d. %-20s %s", marker, i + 1, c.label(), c.soundNotation());
+            if (selected) {
+                writer.print("\u001B[7m"); // reverse video
+                writer.println(line);
+                writer.print("\u001B[0m");
+            } else {
+                writer.println(line);
+            }
+        }
+        writer.println();
+        writer.println("Use ↑/↓ to navigate, Space to replay, Enter to select, Esc to quit");
+    }
+
+    private PronunciationCandidate runFallback(SelectionSession session) {
+        printList(session);
+        System.out.println("Commands: n=down, p=up, space=replay, Enter=select, q=quit");
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
+            while (true) {
+                System.out.print("> ");
+                String line = br.readLine();
+                if (line == null) break;
+                line = line.trim();
+                if (line.isEmpty()) { session.pressEnter(); break; }
+                if (line.equalsIgnoreCase("q")) break;
+                else if (line.equalsIgnoreCase("n")) session.pressDown();
+                else if (line.equalsIgnoreCase("p")) session.pressUp();
+                else if (line.equalsIgnoreCase("space") || line.equals(" ")) session.pressSpace();
+                else System.out.println("Unknown command: " + line);
+                printList(session);
+            }
+        } catch (Exception ignored) {}
```

Why are we ignoring exceptions?